.pre-commit-config.yaml

```diff
@@ -0,0 +1,29 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v6.0.0
+    hooks:
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
+      - id: check-yaml
+      - id: check-added-large-files
+
+  # Hugo site build and test
+  - repo: local
+    hooks:
+      - id: hugo-build-and-test
+        name: Hugo Build and HTMLTest
+        entry: bash -c
+        args:
+          - |
+            set -e
+            echo "Building Hugo site to temporary directory..."
+            hugo build -d /tmp/taranis.ai
```

Comment:
Maybe we could use `.pre-commit-cache/taranis.ai` and add to gitignore as a target.